modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCacheFullTextQueryRebuildSelfTest.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.ignite.cluster.ClusterState;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.spi.communication.tcp.TcpCommunicationSpi;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static java.util.Collections.singleton;
+import static org.apache.ignite.cache.CacheMode.PARTITIONED;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Tests text index rebuild.
+ */
+public class IgniteCacheFullTextQueryRebuildSelfTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        U.resolveWorkDirectory(U.defaultWorkDirectory(), "db", true);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setDataStorageConfiguration(
+            new DataStorageConfiguration().setDefaultDataRegionConfiguration(
+                new DataRegionConfiguration().setPersistenceEnabled(true)
+            )
+        );
+
+        CacheConfiguration cache = new CacheConfiguration(DEFAULT_CACHE_NAME);
+
+        cache.setCacheMode(PARTITIONED);
+        cache.setBackups(1);
+
+        cache.setQueryEntities(singleton(
+            new QueryEntity()
+                .setKeyType(AffinityKey.class.getName())
+                .setValueType(IndexedEntity.class.getName())
+                .setKeyFields(new HashSet<>(Arrays.asList("key", "affKey")))
+                .addQueryField("key", Integer.class.getName(), null)
+                .addQueryField("affKey", Integer.class.getName(), null)
+                .addQueryField("val", String.class.getName(), null)
+                .setIndexes(singleton(
+                    new QueryIndex("val", QueryIndexType.FULLTEXT))
+                )
+        ));
+
+        cfg.setCacheConfiguration(cache);
+
+        TcpCommunicationSpi commSpi = new TcpCommunicationSpi();
+
+        commSpi.setSharedMemoryPort(-1);
+
+        cfg.setCommunicationSpi(commSpi);
+
+        return cfg;
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testFullTextQueryNodeRestart() throws Exception {
+        Ignite crd = startGrids(1);
+
+        crd.cluster().state(ClusterState.ACTIVE);
+
+        IgniteCache<AffinityKey, IndexedEntity> cache = jcache(0, DEFAULT_CACHE_NAME);
```

Comment:
I wonder why you use this specific API, `crd.cache(...)` doesn't work? jcache looks out of place

Reply:
Yep, you are right